R/filterFeatures.R

```diff
@@ -41,8 +53,36 @@ filterFeatures = function(task, method = "randomForestSRC.rfsrc", fval = NULL, p
     threshold = p
   )
 
+  # Caching implementation: @pat-s, Nov 2018
   if (is.null(fval)) {
-    fval = generateFilterValuesData(task = task, method = method, nselect = nselect, ...)$data
+    if (!is.null(cache)) {
+
+      if (!requireNamespace("memoise", quietly = TRUE)) {
+        stop("Package \"memoise\" needed for this function to work. Please install it.",
+             call. = FALSE)
+      }
+
+      # check for user defined cache dir
+      if (class(cache) == "character") {
+        if(!fs::dir_exists(cache)) {
+          fs::dir_create(cache)
+        }
+        cache_dir = cache
+      } else {
+        if(!fs::dir_exists(rappdirs::user_cache_dir("mlr", "mlr-org"))) {
+          fs::dir_create(rappdirs::user_cache_dir("mlr", "mlr-org"))
+        }
+        cache_dir = rappdirs::user_cache_dir("mlr", "mlr-org")
```

Comment:
This should use `get_cache_dir()`. I would also check whether it exists and if it doesn't create it in there.

Reply:
> This should use get_cache_dir().

Completely agree.

> I would also check whether it exists and if it doesn't create it in there.You 

You mean whether the cache is actually written to this directory?

Reply:
Do we really want the extra `fs` dependency just for this?